src/integrationtests/java/com/aws/greengrass/integrationtests/status/EventFleetStatusServiceTest.java

```diff
@@ -214,13 +213,40 @@ void setupKernel(ExtensionContext context) throws Exception {
         kernel.launch();
         assertTrue(fssRunning.await(10, TimeUnit.SECONDS));
         assertTrue(deploymentServiceRunning.await(10, TimeUnit.SECONDS));
+        assertTrue(mainFinished.await(20, TimeUnit.SECONDS));
+        //components with their status already updated will be removed from the set
+        fleetStatusDetailsList.get().forEach(fleetStatusDetails -> fleetStatusDetails.getComponentDetails().
+                forEach(componentStatusDetails -> {
+                    componentNamesToCheck.remove(componentStatusDetails.getComponentName());
+                })
+        );
+        for(FleetStatusDetails f : fleetStatusDetailsList.get()) {
+            for(ComponentDetails s : f.getComponentDetails()) {
+                componentNamesToCheck.remove(s.getComponentName());
+            }
+        }
+        assertTrue(mainFinished.await(20, TimeUnit.SECONDS));
+        // 2 additional messages will be added to fleetStatusDetailsList. After Nucleus launch, components up and running and then main stop and finish
     }
 
     @AfterEach
     void afterEach() {
         kernel.shutdown();
     }
 
+    @Test
+    void GIVEN_Nucleus_launch_THEN_fss_send_launch_message_and_running_message() throws Exception {
```

Comment:
overall, i think this test should assert that a device sends a nucleus launch message first and then component change messages

Reply:
Updated